arelle/plugin/validate/NL/rules/fr_nl.py

```diff
@@ -568,6 +568,32 @@ def rule_fr_nl_4_02(
         )
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[
+        DISCLOSURE_SYSTEM_NT16,
+        DISCLOSURE_SYSTEM_NT17,
+        DISCLOSURE_SYSTEM_NT18
+    ],
+)
+def rule_fr_nl_5_03(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation] | None:
+    """
+    FR-NL-5.03: An XBRL instance document MUST NOT contain empty item concepts
+    """
+    for fact in val.modelXbrl.facts:
+        if not fact.xValue and not fact.concept.instanceOfType("xbrli:stringItemType"):
```

Comment:
This check appears to be implementing "if an element is empty, it must be a string", but it seems like the rule is saying "don't allow empty items, which only matters for string item types", i.e.:
```
if fact.concept.instanceOfType(qnXbrliStringItemType) and not fact.xValue:
```

It looks like the argument to `instanceOfType` should be `QName`, not `str`?  It would be nice to add a type hint at some point.

Reply:
I confirmed this with Anna. If a fact is tagged with a concept that is a derivative of string Item type and it has no value then this validation should fire.